File: frontend/src/api/prompts.js
import service from './index'

/**
 * Get all current prompt settings
 */
export const getPrompts = async () => {
  try {
    return await service.get('/api/prompts')
  } catch (err) {
    return { success: false, error: err.message }
  }
}

/**
 * Update prompt settings
 * @param {Object} data - prompt data keyed by category
 */
export const updatePrompts = async (data) => {
  try {
    return await service.put('/api/prompts', data)
  } catch (err) {
    return { success: false, error: err.message }
  }
}

/**
 * Reset all prompts to default values
 */
export const resetPrompts = async () => {
  try {
    return await service.post('/api/prompts/reset')
  } catch (err) {
    return { success: false, error: err.message }
  }
}

File: backend/app/api/prompts.py
"""
Prompts API - CRUD for prompt templates
"""

from flask import jsonify, request
from . import prompts_bp
from ..prompts.prompt_loader import get_all_prompts, save_prompts, reset_prompts


@prompts_bp.route('', methods=['GET'])
def get_prompts():
    """Get all prompt templates."""
    try:
        prompts = get_all_prompts()
        return jsonify({'success': True, 'data': prompts})
    except Exception as e:
        return jsonify({'success': False, 'error': str(e)}), 500


@prompts_bp.route('', methods=['PUT'])
def update_prompts():
    """Update prompt templates."""
    try:
        data = request.get_json()
        if not data:
            return jsonify({'success': False, 'error': 'No data provided'}), 400
        save_prompts(data)
        return jsonify({'success': True, 'message': 'Prompts updated'})
    except Exception as e:
        return jsonify({'success': False, 'error': str(e)}), 500


@prompts_bp.route('/reset', methods=['POST'])
def reset_to_defaults():
    """Reset prompts to defaults."""
    try:
        reset_prompts()
        prompts = get_all_prompts()
        return jsonify({'success': True, 'data': prompts, 'message': 'Reset to defaults'})
    except Exception as e:
        return jsonify({'success': False, 'error': str(e)}), 500
